src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,123 @@
+    }
+
+    @Test
+    void register_invalidLogin_notOk() {
+        testUser.setAge(20);
+        testUser.setPassword("password");
+        testUser.setLogin(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setLogin("");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setLogin("abc");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setLogin("abcde");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_validLogin_Ok() {
+        testUser.setAge(20);
+        testUser.setPassword("password");
+        testUser.setLogin("abcdef");
+        assertEquals(testUser, registrationService.register(testUser));
+        testUser = new User();
+        testUser.setAge(20);
+        testUser.setPassword("password");
+        testUser.setLogin("11111111111111111");
+        assertEquals(testUser, registrationService.register(testUser));
+    }
+
+    @Test
+    void register_invalidPassword_notOk() {
+        testUser.setAge(20);
+        testUser.setLogin("validLogin");
+        testUser.setPassword(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setPassword("");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setPassword("pas");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setPassword("passw");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_validPassword_Ok() {
+        testUser.setAge(20);
+        testUser.setLogin("Login1");
+        testUser.setPassword("validP");
+        assertEquals(testUser, registrationService.register(testUser));
+        testUser = new User();
+        testUser.setAge(20);
+        testUser.setLogin("Login2");
+        testUser.setPassword("fullyValidPassword");
+        assertEquals(testUser, registrationService.register(testUser));
+    }
+
+    @Test
+    void register_invalidAge_notOk() {
+        testUser.setLogin("validLogin");
+        testUser.setPassword("validPass");
+        testUser.setAge(-100);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setAge(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setAge(3);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
+        testUser.setAge(17);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(testUser));
```

Comment:
Consider adding a test case for an age that is exactly the minimum required age to ensure boundary conditions are handled correctly.